libshaderc_spvc/src/spvc.cc

```diff
@@ -515,7 +625,240 @@ shaderc_spvc_status shaderc_spvc_needs_buffer_size_buffer(
+             .get(spv::DecorationBinding)) {
+      context->messages.append(
+          "Unable to get binding decoration for shader resource\n");
+      return shaderc_spvc_status_internal_error;
+    }
+    uint32_t binding_decoration =
+        compiler->get_decoration(shader_resource.id, spv::DecorationBinding);
+    bindings->binding = binding_decoration;
+
+    if (!compiler->get_decoration_bitset(shader_resource.id)
+             .get(spv::DecorationDescriptorSet)) {
+      context->messages.append(
+          "Unable to get descriptor set decoration for shader resource\n");
+      return shaderc_spvc_status_internal_error;
+    }
+    uint32_t descriptor_set_decoration = compiler->get_decoration(
+        shader_resource.id, spv::DecorationDescriptorSet);
+    bindings->set = descriptor_set_decoration;
+
+    bindings->id = shader_resource.id;
+    bindings->base_type_id = shader_resource.base_type_id;
+
+    switch (binding_type) {
+      case shaderc_spvc_binding_type_sampled_texture: {
+        spirv_cross::SPIRType::ImageType imageType =
+            compiler->get_type(bindings->base_type_id).image;
+        spirv_cross::SPIRType::BaseType textureComponentType =
+            compiler->get_type(imageType.type).basetype;
+        bindings->multisampled = imageType.ms;
+        bindings->texture_dimension = spirv_dim_to_texture_view_dimension(
+            imageType.dim, imageType.arrayed);
+        bindings->texture_component_type =
+            spirv_cross_base_type_to_texture_format_type(textureComponentType);
+        bindings->binding_type = binding_type;
+      } break;
+      case shaderc_spvc_binding_type_storage_buffer: {
+        // Differentiate between readonly storage bindings and writable ones
+        // based on the NonWritable decoration
+        spirv_cross::Bitset flags =
+            compiler->get_buffer_block_flags(shader_resource.id);
+        if (flags.get(spv::DecorationNonWritable)) {
+          bindings->binding_type =
+              shaderc_spvc_binding_type_readonly_storage_buffer;
+        } else {
+          bindings->binding_type = shaderc_spvc_binding_type_storage_buffer;
+        }
+      } break;
+      default:
+        bindings->binding_type = binding_type;
+    }
+    bindings++;
+  }
+
+  return shaderc_spvc_status_success;
+}
+
+shaderc_spvc_status shaderc_spvc_get_input_stage_location_info(
+    const shaderc_spvc_context_t context,
+    shaderc_spvc_resource_location_info* locations, size_t* location_count) {
+  if (!context) return shaderc_spvc_status_missing_context_error;
+
+  if (!context->cross_compiler) {
+    context->messages.append(
+        "Invoked get_input_locations without an initialized compiler\n");
+    return shaderc_spvc_status_uninitialized_compiler_error;
+  }
+  auto* compiler = context->cross_compiler.get();
+
+  if (!location_count) {
+    context->messages.append(
+        "Invoked get_input_locations without a valid location_count param\n");
```

Comment:
nit: get_input_stage_location_info ?

Reply:
Yup, symbol rename in my ide doesn't catch random strings that use part of the function name